### PR TITLE
Make LMS models great again

### LINKED ISSUE #282: Models with primary key != id can't be `PATCH`ed
It is a django ~~bug~~ feature. If a model like this

```python
class Rank(models.Model):
    rank = models.CharField(primary_key=True, max_length=30)
```
 is `PATCH`ed, new instance is created and the old one remains. This could be fixed by introducing ids to the model and making them primary keys, whereas fields that were previously primary should be marked as unique.

### LINKED ISSUE #345: Bump `populate` for easier development
- [x] Link students with users (auth).
- [x] Link teachers with users (auth).
- [x] Add more milgroup leaders (uniform updates).
- [x] Add more applicants (registration approvements).
- [x] Populate `tgbot` user (tgbot interaction).
- [x] Add `lesson-subjects.get.all` to all groups
- [x] Add `teacher.patch.self` to teachers group
- [x] Add `encouragement.delete.self` to teacher group
- [x] Add `punishments.delete.self` to teacher group
- [x] Add `marks.*.milfaculty` to teacher group
- [x] Add `marks.put.milfaculty` to milfaculty_head group
- [x] Remove `punishments/encouragements.post.milgroup` + filter student and teacher
- [x] Left `PUT` only for `marks`

### PR DESCRIPTION
**Pull request by TmLev**
Resolves #282, closes #345.

### CODE REVIEW

File: back-end/src/auth/populate/permissions.py [hunk 1]
```diff
@@ -30,29 +31,48 @@ def get_student_permissions():
 
 def get_teacher_permissions():
     values = [
+        # DMS permissions
```
Comment (TmLev) [inline back-end/src/auth/populate/permissions.py:34]: WIP.

**Review by gakhromov — DISMISSED** [review]
LGTM

File: back-end/src/lms/views/marks.py [hunk 2]
```diff
@@ -255,8 +257,8 @@ def get(self, request: Request) -> Response:
         if scope == Permission.Scope.SELF:
             filter_kwargs = {"user": request.user}
         else:
-            filter_kwargs = {"milgroup": milgroup}
-
+            filter_kwargs = {"milgroup": milgroup_id}
+        print(filter_kwargs)
```
Comment (TmLev) [inline back-end/src/lms/views/marks.py:261]: Remove.

File: front-end/src/utils/enums.js [hunk 3]
```diff
@@ -43,3 +48,29 @@ export const ABSENCE_STATUSES = {
...
+  FI: "Зачет",
+  EX: "Экзамен",
+};
+export const HEAD_DRESSES = {
+  CA, HA,
+};
+export const OUTERWEARS = {
+  JA: "",
+  PC,
+};
```
Comment (TmLev) [inline front-end/src/utils/enums.js:76]: Values are missing.

**Review by gakhromov — APPROVED** [review]
Nice job @TmLev @gakhromov @askatsevalov !

**Pull request merged**